mmtrack/datasets/pipelines/transforms.py

```diff
@@ -134,13 +137,268 @@ def __call__(self, results):
+
+        return img_crop_padded, resize_factor, pdding_mask
+
+    def generate_box(self,
+                     bbox_gt,
+                     bbox_cropped,
+                     resize_factor,
+                     output_size,
+                     normalize=False):
+        """Transform the box coordinates from the original image coordinates to
+        the coordinates of the cropped image.
+
+        Args:
+            bbox_gt (ndarray): of shape (4, ) in [x1, y1, x2, y2] format.
+            bbox_cropped (ndarray): of shape (4, ) in [x1, y1, x2, y2] format.
+            resize_factor (float): the ratio of original image scale to cropped
+                image scale.
+            output_size (float): the size of output image.
+            normalize (bool): whether to normalize the output box.
+                Default to True.
+
+        Returns:
+            ndarray: generated box of shape (4, ) in [x1, y1, x2, y2] format.
+        """
+        assert output_size > 0
+        bbox_gt_center = (bbox_gt[0:2] + bbox_gt[2:4]) * 0.5
+        bbox_cropped_center = (bbox_cropped[0:2] + bbox_cropped[2:4]) * 0.5
+
+        bbox_out_center = (output_size - 1) / 2. + (
+            bbox_gt_center - bbox_cropped_center) * resize_factor
+        bbox_out_wh = (bbox_gt[2:4] - bbox_gt[0:2]) * resize_factor
+        bbox_out = np.concatenate((bbox_out_center - 0.5 * bbox_out_wh,
+                                   bbox_out_center + 0.5 * bbox_out_wh),
+                                  axis=-1)
+
+        return bbox_out / output_size if normalize else bbox_out
+
+    def __call__(self, results):
+        """Call function. For each dict in results, crop image like Stark did.
+
+        Args:
+            results (list[dict]): list of dict from
+                :obj:`mmtrack.base_sot_dataset`.
+
+        Returns:
+            List[dict]: list of dict that contains cropped image and
+                the corresponding groundtruth bbox.
+        """
+        outs = []
+        for i, _results in enumerate(results):
+            image = _results['img']
+            gt_bbox = _results[_results.get('bbox_fields', [])[0]][0]
+            jittered_bboxes = _results['jittered_bboxes'][0]
+            crop_img, resize_factor, att_mask = self.crop_like_stark(
+                image, jittered_bboxes, self.crop_size_factor[i],
+                self.output_size[i])
+
+            generated_bbox = self.generate_box(
+                gt_bbox,
+                jittered_bboxes,
+                resize_factor,
+                self.output_size[i],
+                normalize=False)
+
+            generated_bbox = generated_bbox[None]
+
+            _results['img'] = crop_img
+            if 'img_shape' in _results:
+                _results['img_shape'] = crop_img.shape
+            _results[_results.get('bbox_fields', [])[0]] = generated_bbox
+            _results['seg_fields'] = ['att_mask']
+            _results[_results.get('seg_fields', [])[0]] = att_mask
```

Comment:
Do not use `_results.get('seg_fields', [])[0]` to get the mask.
The `xxx_field` is used for preprocessing some variables with the same type, do not use hard code `[0]` to get the specific variable.
You should use `_results['att_mask'] = att_mask` to record the mask

Reply:
Then please use `_results['att_mask'] = att_mask` to record the mask